analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CloudNative/AzureFunctionsCatchExceptions.cs

```diff
@@ -0,0 +1,534 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+
+namespace Microsoft.Azure.WebJobs
+{
+    public class FunctionNameAttribute : Attribute { public FunctionNameAttribute(string name) { } }  // FIXME: Fake, remove before merging
+}
+
+public class AnotherAttribute : Attribute
+{
+    public AnotherAttribute(string name) { }
+}
+
+public static class Functions
+{
+    private const int ConstantInt = 42;
+
+    delegate void VoidDelegate();
+
+    public static void NotAnAzureFunction()     // Compliant
+    {
+        DoSomething();
+    }
+
+    [Another("Something")]
+    public static void WithAnotherAttribute()   // Compliant
+    {
+        DoSomething();
+    }
+
+    [FunctionName("Sample")]
+    public static void NoTryCatch_Body(int arg)         // Noncompliant {{Wrap Azure Function body in try/catch block.}}
+    //                 ^^^^^^^^^^^^^^^
+    {
+        DoSomething();
+    }
+
+    [FunctionName("Sample")]
+    public static void NoTryCatch_Arrow() =>            // Noncompliant
+        DoSomething();
+
+    [FunctionName("Sample")]
+    public static async Task<string> NoTryCatchAsync()  // Noncompliant
+    {
+        return DoSomething();
+    }
+
```

Comment:
Consider adding a test case with a ConditionalAccessExpression like `SomeInstance?.DoSomething();` and `GetSomeInstance()?.DoSomething();`